cpp/src/fil/treelite_import.cu

```diff
@@ -69,32 +69,64 @@ int tree_root(const tl::Tree<T, L>& tree)
   return 0;  // Treelite format assumes that the root is 0
 }
 
-template <typename T, typename L>
-inline int max_depth(const tl::Tree<T, L>& tree)
+using empty = struct {
+};
+
+template <typename T,
+          typename L,
+          typename DescentAccumulator,
+          typename InnerFunc,
+          typename LeafFunc>
+inline void walk_tree(const tl::Tree<T, L>& tree,
+                      DescentAccumulator desc_acc,
+                      InnerFunc inner_func,
+                      LeafFunc leaf_func)
```

Comment:
Consider having a default `leaf_func` that does nothing (and a matching default `LeafFunc` for it).